pkgs/tools/security/vaultwarden/webvault.nix

```diff
@@ -1,19 +1,63 @@
-{ lib, stdenv, fetchurl, nixosTests }:
+{ lib
+, buildNpmPackage
+, fetchFromGitHub
+, git
+, nixosTests
+, nodejs-16_x
+, python3
+}:
 
-stdenv.mkDerivation rec {
+let
+  buildNpmPackage' = buildNpmPackage.override { nodejs = nodejs-16_x; };
+
+  version = "2023.1.1";
+
+  bw_web_builds = fetchFromGitHub {
+    owner = "dani-garcia";
+    repo = "bw_web_builds";
+    rev = "v${version}";
+    hash = "sha256-JLl6rqEOp940BHnqQ7AFiMvRkIG02Hwu9oQ6fe1oexQ=";
+  };
+in buildNpmPackage' {
   pname = "vaultwarden-webvault";
-  version = "2022.12.0";
+  inherit version;
 
-  src = fetchurl {
-    url = "https://github.com/dani-garcia/bw_web_builds/releases/download/v${version}/bw_web_v${version}.tar.gz";
-    hash = "sha256-QC3/aqIF2NdJPHmwUbvJR62wsUGBrgsHJCyqBJ/0gMc=";
+  src = fetchFromGitHub {
+    owner = "bitwarden";
+    repo = "clients";
+    rev = "web-v${version}";
+    hash = "sha256-hA/j/8JQqw9Y8EyUOZYrZ59MsfiSRhaVUf1yWs32Hyc=";
   };
 
-  buildCommand = ''
-    mkdir -p $out/share/vaultwarden/
-    cd $out/share/vaultwarden/
-    tar xf $src
-    mv web-vault vault
+  npmDepsHash = "sha256-cElToZplSYWfSxi2wfFehVvk0SHbLXC8zeRWG9z9hbk=";
+
+  postPatch = ''
+    cp ${bw_web_builds}/scripts/apply_patches.sh .
+    substituteInPlace apply_patches.sh \
```

Comment:
Nit: any reason to not just stick it in $PATH instead? Might be a bit more resilient.

Reply:
True, I'll do that.
It would have been even nicer to put it in `nativeBuildInputs` but that isn't honored by `npmFetchDeps`.